packages/smooth_app/lib/pages/scan/smooth_barcode_scanner_awesome.dart

```diff
@@ -0,0 +1,188 @@
+        width: 1024,
+        maxFramesPerSecond: 5,
+      ),
+      builder: (
+        CameraState cameraState,
+        PreviewSize previewSize,
+        Rect previewRect,
+      ) {
+        // Return your UI (a Widget)
+        return cameraState.when(
+          onPreparingCamera: (PreparingCameraState state) =>
+              const Center(child: CircularProgressIndicator()),
+          onPhotoMode: (PhotoCameraState state) => BarcodeScannerViewWidget(
+            onScan: onScan,
+            cameraState: cameraState,
+            previewSize: previewSize,
+            previewRect: previewRect,
+          ),
+          onVideoRecordingMode: (VideoRecordingCameraState state) =>
+              BarcodeScannerViewWidget(
+            onScan: onScan,
+            cameraState: cameraState,
+            previewSize: previewSize,
+            previewRect: previewRect,
+          ),
+          onVideoMode: (VideoCameraState state) => BarcodeScannerViewWidget(
+            onScan: onScan,
+            cameraState: cameraState,
+            previewSize: previewSize,
+            previewRect: previewRect,
+          ),
+        ) as Widget;
+      },
+    );
+  }
+
+  Future<void> _processImageBarcode(AnalysisImage img) async {
+    final Size imageSize = Size(img.width.toDouble(), img.height.toDouble());
+    final InputImageRotation imageRotation =
+        InputImageRotation.values.byName(img.rotation.name);
+
+    final List<InputImagePlaneMetadata> planeData = img.planes.map(
+      (ImagePlane plane) {
+        return InputImagePlaneMetadata(
+          bytesPerRow: plane.bytesPerRow,
+          height: img.height,
+          width: img.width,
+        );
+      },
+    ).toList();
+
+    final InputImage inputImage;
+
+    if (Platform.isIOS) {
+      final InputImageData inputImageData = InputImageData(
+        size: imageSize,
+        imageRotation: imageRotation,
+        inputImageFormat: _inputImageFormat(img.format),
+        planeData: planeData,
+      );
+
+      final WriteBuffer allBytes = WriteBuffer();
+      for (final ImagePlane plane in img.planes) {
+        allBytes.putUint8List(plane.bytes);
+      }
+      final Uint8List bytes = allBytes.done().buffer.asUint8List();
+
+      inputImage =
+          InputImage.fromBytes(bytes: bytes, inputImageData: inputImageData);
+    } else {
+      inputImage = InputImage.fromBytes(
+        bytes: img.nv21Image!,
+        inputImageData: InputImageData(
+          imageRotation: imageRotation,
+          inputImageFormat: InputImageFormat.nv21,
+          planeData: planeData,
+          size: Size(img.width.toDouble(), img.height.toDouble()),
+        ),
+      );
+    }
+
+    try {
+      final List<Barcode> recognizedBarCodes =
+          await _barcodeScanner.processImage(inputImage);
+      for (final Barcode barcode in recognizedBarCodes) {
+        debugPrint('Barcode: [${barcode.format}]: ${barcode.rawValue}');
+
+        onScan(barcode.rawValue.toString());
```

Comment:
There nothing more elegant than `.rawValue.toString()`?

Reply:
Dont think so, that I just took over from the docs